### ISSUE TITLE
[Suggestion] Coping with subdirectories of webserver document root.

### ISSUE THREAD

**Issue by gbonnema** [zendframework/zend-expressive-skeleton#108]
Due to the issue #107 I wondered whether Expressive could have some option to deal with subdirectories. 

For instance using environment dependant configuration options. In my case it is only necessary for the test environment where a developer has multiple websites under one document root, and no way to change the document root (i.e. no root access).

From the cookbook (http://zendframework.github.io/zend-expressive/cookbook/using-a-base-path/) I gather that the problem is well know and well understood. 

Would it be a good idea to incorporate such an option into Expressive? Or at least enter into the standard expressive docs how to accomplish this?

**Comment by xtreamwayz** [comment]
That URL is the location of the official standard expressive docs. Well, sort of. It's actually https://docs.zendframework.com/zend-expressive/cookbook/using-a-base-path/ but they both point to the same files.

There are several other ways to accomplish what you want for developing multiple sites locally:

- Use a `.htaccess` file and some rewrite rules.
- Use vagrant or docker and build an exact copy of your server.
- Use the host file. Add something like my-site.dev and my-site2.dev to the host file and make apache point to the right source code.

**Comment by gbonnema** [comment]
=== Quick start ===
I am new to Zend (and Zend 3). For me the cookbook was too far from the "quick start" instructions. I got stuck at point 5 (Start a webserver) when I replaced the internal server (php -S 0.0.0.0) by a subdirectory of my company's testserver.  Point 5 in the quickstart might be a good location for a quick remark and some pointers (f.i. to the cookbook) for ppl that want to use subdirectories on testservers.

=== Middleware ===
The other part of the question I had is whether the expressive skeleton could automatically take care of subdirs. From the cookbook I understand that .htaccess rewrite rules are only part of the solution. The middleware needs adapted to take care of the rest. 

P.S. I wish I had the knowledge to propose a code change, but I am still wrestling with the Zend3 contents, so for now that is out.
